kato/kato-aws/src/main/groovy/com/netflix/spinnaker/kato/aws/deploy/AutoScalingWorker.groovy

```diff
@@ -117,6 +118,7 @@ class AutoScalingWorker {
       classicLinkVpcId: classicLinkVpcId,
       instanceType: instanceType,
       keyPair: keyPair,
+      base64UserData: base64UserData,
```

Comment:
I think this is the right spot to compute and set the userData - will need a bit of reworking to pull some of the insides out of LaunchConfigurationBuilder (e.g. the name needs to be known at the time userdata templating happens) but this can do the merge of userDataProvider value and deploy description value and the createLaunchConfiguration can just set this resulting field on the newly created launch configuration